components/service/glean/src/main/java/mozilla/components/service/glean/utils/FreshBootDetector.kt

```diff
@@ -4,92 +4,79 @@
 
 package mozilla.components.service.glean.utils
 
+import android.content.BroadcastReceiver
 import android.content.Context
-import android.os.Build
-import android.provider.Settings
+import android.content.Intent
+import android.content.SharedPreferences
 import android.support.annotation.VisibleForTesting
-import java.io.File
-import java.io.IOException
-import java.util.UUID
+import mozilla.components.support.base.log.logger.Logger
+
+/*
+ * Receives the BOOT_COMPLETED intent.
+ *
+ * Sets a pref when the device first boots up.
+ */
+class BootCompletedReceiver : BroadcastReceiver() {
```

Comment:
@mdboom the new permission requirement shuts me down a bit on this approach. What do you think of approaching this like "always send any pending chunk of data" for now? I really don't want us to block on the boot detection mechanism: we could file a bug, track that as future follow-up work and document the fact that we might generate pings every time we start in `events.md`.

Reply:
One data point: the existing telemetry library [requests this permission](https://github.com/mozilla-mobile/android-components/blob/c8583395fc902aa964e101fa20c652d16c7fc02d/components/service/telemetry/src/main/AndroidManifest.xml#L8), though I'm not sure where it's actually used.

Reply:
Ah, I see. Then it's fine for me.